### ISSUE TITLE
Ingress and dns resolution fails due to networkpolicy

### ISSUE THREAD

**Issue by ShylajaDevadiga** [rancher/rke2#305]
Version:
rke2 -v
rke2 version v1.18.4-beta17+rke2 (4c90fcab1e478d957d59e7ac9e4ebc1083330b79)

OS Version:
cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.8 (Maipo)

Issue: 
Ingress unable to handle request 
```
curl -H  "host:foo1.bar.com" http://3.14.129.74/name.html
<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.17.8</center>
</body>
</html>
```
Ingress works by deleting networkpolicy.

```
kubectl delete networkpolicy --all -n kube-system && kubectl delete networkpolicy --all -n default

curl -H  "host:foo1.bar.com" http://3.14.129.74/name.html 
testk8s1-7rqkm
```
PR to fix networkpolicy issue rancher/rke2#299

**Comment by davidnuzik** [comment]
This issue is just for tracking testing (and any issues found out of testing) because https://github.com/rancher/rke2/issues/302 is broken.

**Comment by davidnuzik** [comment]
This is not a duplicate of https://github.com/rancher/rke2/issues/297 for testing #299 ?

**Comment by ShylajaDevadiga** [comment]
Validated issue is resolved in rke2 version v1.18.8-beta19+rke2 
Node OS
Rhel 7.8
SELinux enabled
Ingress and dns resolution works
```
➜  ~ curl -H "host:foo1.bar.com" 3.137.166.50/name.html
testk8s1-7fk49
➜  ~ curl -H "host:foo1.bar.com" 3.137.166.50/name.html
testk8s1-mpb5b

k exec -it dnsutils --kubeconfig=/etc/rancher/rke2/rke2.yaml -- nslookup kubernetes.default
Server:		10.43.0.10
Address:	10.43.0.10#53

Name:	kubernetes.default.svc.cluster.local
Address: 10.43.0.1
```
